testing/unittests/cryptotests.cpp

```diff
@@ -347,6 +385,65 @@ class CryptoUnitTest : public CppUnit::TestFixture
         _pkeEncryptDecryptTest();
     }
 
+    void _pkeEncryptDecryptPassphraseTest()
+    {
+        try
+        {
+            // create random data
+            MemoryBuffer toEncryptMb;
+            fillRandomData(64, toEncryptMb);
+
+            MemoryBuffer pkeMb;
+            MemoryBuffer decryptedMb;
+
+            /////////////////////////////////////
+            //PKE tests using a passphrase string
+            /////////////////////////////////////
+
+            Owned<CLoadedKey> publicKey = loadPublicKeyFromMemory(pubKeyPassphrase);
+            Owned<CLoadedKey> privateKey = loadPrivateKeyFromMemory(privKeyPassphrase, "ThisIsMyPassphrase");
+
+            publicKeyEncrypt(pkeMb, toEncryptMb.length(), toEncryptMb.bytes(), *publicKey);
+            privateKeyDecrypt(decryptedMb, pkeMb.length(), pkeMb.bytes(), *privateKey);
+
+            ASSERT(toEncryptMb.length() == decryptedMb.length());
+            ASSERT(0 == memcmp(toEncryptMb.bytes(), decryptedMb.bytes(), toEncryptMb.length()));
+
+            /////////////////////////////////////
+            //PKE tests using a passphrase buffer
+            /////////////////////////////////////
+
+            Owned<CLoadedKey> publicKeyPassphrase = loadPublicKeyFromMemory(pubKeyPassphrase);
+            Owned<CLoadedKey> privateKeyPassphrase = loadPrivateKeyFromMemory(privKeyPassphrase, 18, "ThisIsMyPassphrase");
```

Comment:
to prove the point, that this was to support binary pass phrases, can you change this so that it's a binary pass phrase with nulls in middle of it?